### PR TITLE
Add ability to click select and checkbox & radio inside Dropdown.

### LINKED ISSUE #17656: Dropdown should ignore clicks on <select>s (and <label>s?) within the dropdown
Hello.
I tried to use dropdown plugin like a special area for options (some radio buttons) and found a strange behavior. You can see it here (try to click on select tag or labels for radio buttons): http://jsbin.com/jorota/edit?html,output

There is a simple check in bootstrap.js in this plugin
```javascript
/input|textarea/i.test(e.target.tagName)
```

this behavior can be fixed by adding new tags through the '|'
```javascript
/input|textarea|select|label/i.test(e.target.tagName)
```

is it possible to add this fix to release? or this is not a type of regular using of this plugin? but in this case: why there is 'textarea' in condition? :)

thank you :) and sorry for my bad english

### PR DESCRIPTION
**Pull request by villeristi**
Add ability to click select and Bootstrap-style checkbox- & radio-divs inside dropdown without loosing its visibility. Fixes #17656.

### CODE REVIEW

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 287c57a3e369919bfd365752054ef2657cbb6cdd
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/82643135

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

File: js/dropdown.js [hunk 1]
```diff
@@ -44,7 +44,7 @@
 
       if (!$parent.hasClass('open')) return
 
-      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return
+      if (e && e.type == 'click' && (/input|select|textarea/i.test(e.target.tagName) || /checkbox|radio/i.test(e.target.className)) && $.contains($parent[0], e.target)) return
```
Comment (cvrebert) [inline js/dropdown.js]: Seems like this should be `type` instead of `className`. Folks might be using vanilla non-Bootstrap checkboxes.

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 2be6da68155d17ec7b5c2516d5c36ae225a98fc3
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/82689529

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by mdo** [comment]
[**Bootstrap 3 is no longer being officially developed or supported.**](https://github.com/twbs/bootstrap/issues/20631)

All work has moved onto our next major release, v4. As such, this issue or pull request is being closed as a "won't fix." For additional help and support, we recommend [utilizing our community resources](https://github.com/twbs/bootstrap/blob/master/README.md#community). Thanks for your understanding, and see you on the other side of v4!

<3,
@mdo and team

**Pull request closed without merging**